libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -417,9 +417,6 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 
 		_variable.annotation().type = type;
 	}
-	else if (!_variable.canHaveAutoType())
```

Comment:
Can you instead put an assert at the very beginning of this function and unindent everything by one step?